presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -597,6 +597,10 @@ private static boolean typesMatchForInsert(Iterable<Type> tableTypes, Iterable<T
             Type tableType = tableTypesIterator.next();
             Type queryType = queryTypesIterator.next();
 
+            if (isNullType(queryType)) {
+                continue;
```

Comment:
Is there a check available to see if the table allows null values to be inserted?